src/v/storage/disk_log_impl.h

```diff
@@ -60,7 +60,7 @@ class disk_log_impl final : public log::impl {
     disk_log_impl(const disk_log_impl&) = delete;
     disk_log_impl& operator=(const disk_log_impl&) = delete;
 
-    ss::future<> close() final;
+    ss::future<std::optional<ss::sstring>> close() final;
```

Comment:
Can you add a comment to this function, describing what the function does and the meaning of the return value?